ListAverages_v2_Homework1.py

```diff
@@ -0,0 +1,37 @@
+# this solution is to satisfy the following criteria: sort without using sort() and catch exceptions
+import random # import random module to generate random numbers
+
+# create a list
+random_list = [] # start with empty random list
+for n in range (0,100): # we need 100 elements in a list
+    number = random.randint(0,1000) # generate random integer from 0 to 1000 for each n in range 0-100
+    random_list.append(number) # add every random integer generated to the list
+
+# sort the list created
+for index in range(len(random_list)): # list has len(list) elements; we want to iterate through each element and compare it to others
+    for compared_index in range(index + 1, len(random_list)): # compared index = position of all other elements starting from the next element
+        if random_list[index] > random_list[compared_index]: # compare value of the index to the value of each next index
+            random_list[index], random_list[compared_index] = random_list[compared_index], random_list[index] # if current value is greater than compared value assign their values to each other indexes (swap their positions in a list)
```

Comment:
The good realization of the Bubble sort! 

Some additional info for you:

Of course, the performance is not so important if you have only 1000 numbers. But generally, Bubble sort isn't the best choice. The comparison of most popular sorting algorithms: 
![image](https://user-images.githubusercontent.com/34323554/105741151-a7a4ae80-5f42-11eb-8326-4e81460ad817.png)

For example, countingSort is very fast for large N, but requires a range sufficient to handle the numbers involved. For small differences between minimum(a[]), the smallest element in a[] and maximum(a[]), the largest element in a[], this shouldn't pose a problem. It is not a memory-efficient method otherwise. Typical performance is O(n). Its usefulness is limited to integer-domain arrays.

BubbleSort is useful ONLY **when arrays are nearly sorted.** Its performance is terrible on **large unordered arrays.**

InsertionSort is useful ONLY when arrays are nearly sorted and is nearly O(n) in the case of nearly-sorted lists. Like BubbleSort, it is stable. It is also used as part of IntroSort and optimized versions of MergeSort and other hybrid algorithms.

More details with examples: https://medium.com/@george.seif94/a-tour-of-the-top-5-sorting-algorithms-with-python-code-43ea9aa02889

Reply:
Thank you so much! Great feedback, very useful!